arch/arc/core/thread.c

```diff
@@ -84,12 +130,34 @@ void _new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	 * value.
 	 */
 #ifdef CONFIG_ARC_STACK_CHECKING
-	pInitCtx->status32 = _ARC_V2_STATUS32_SC | _ARC_V2_STATUS32_E(_ARC_V2_DEF_IRQ_LEVEL);
+	pInitCtx->status32 = _ARC_V2_STATUS32_SC |
+		 _ARC_V2_STATUS32_E(_ARC_V2_DEF_IRQ_LEVEL);
 	thread->arch.stack_base = (u32_t) stackEnd;
 #else
 	pInitCtx->status32 = _ARC_V2_STATUS32_E(_ARC_V2_DEF_IRQ_LEVEL);
 #endif
 
+#if CONFIG_USERSPACE
+	/*
+	 * enable US bit, US is read as zero in user mode. This will allow use
```

Comment:
might be good to mention why we are enabling (if I remember right, to enable a quick check of the current CPU privilege mode?) since the sleep mode thing is a side effect.

Reply:
Yes, it's to enable a quick check of the current CPU privilege mode. The side effect is that the processor can be put into sleep in User mode.  As discussed with Ruud,  it's acceptable and not worse than the forever loop in user mode

Reply:
Right, what I was saying was that I would like this particular comment to be amended with this information as it's not obvious from looking at the code.